### ISSUE TITLE
feat(radtran): quality flags, screening, and information-content diagnostics

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/plumax#97]
## Problem / Request
Retrievals need a screening layer (cloud/cirrus, AOD ceiling, SZA/VZA limits, χ² and convergence flags) and information-content diagnostics (DOFs, averaging-kernel row sums, Shannon information) — both ☐ on the RTM design page (`radtran.quality`, `radtran.diagnostics`). The design's v1 decision is to trust the L1 cloud mask and enforce a strict clear-sky AOD < 0.2 ceiling; the diagnostics are what catch retrieval-side regressions that mean-error tests miss.

## User Story
> As a user retrieving a scene, I want a per-pixel quality mask with named reasons and a DOFs map, so I can tell an uninformative pixel from a bad one and Tier IV can aggregate the flags.

## Proposed API
```python
# src/plumax/radtran/quality.py
class Flag(IntFlag): CLOUD = 1; AOD = 2; GEOMETRY = 4; CHI2 = 8; NOT_CONVERGED = 16; SATURATED = 32
@dataclass(frozen=True)
class QualityThresholds: aod_max: float = 0.2; sza_max_deg: float = 70.0; chi2_max: float = 3.0
def quality_flags(result: OEResult, *, geometry: ObservationGeometry, l1_cloud_mask, aod, thresholds=QualityThresholds()) -> Int[Array, "..."]: ...

# src/plumax/radtran/diagnostics.py
def degrees_of_freedom(A: jax.Array) -> float: ...           # trace(A)
def shannon_information(S_a, S_hat) -> float: ...             # ½ log det(S_a S_hat⁻¹)
def theoretical_max_dofs(instrument: InstrumentSpec, srf: SpectralResponseFunction, snr: float) -> float: ...
```

## Implementation Steps
- [ ] Add `radtran/quality.py` and `radtran/diagnostics.py`; reuse `assimilation.diagnostics.degrees_of_freedom_for_signal` where the maths coincides
- [ ] `matched_filter` path gets the same `Flag` bits it can compute (geometry, saturation)
- [ ] Document the "mask (default) vs down-weight via inflated R" choice for Tier IV

## Definition of Done
- [ ] DOFs vs theoretical maximum tracks SNR/SZA monotonically on synthetic radiances
- [ ] A pixel above the AOD ceiling is flagged and excluded from `fuse_observations` when passed through `coupled.Instrument`
- [ ] `make test && make lint && make typecheck` green

## Testing
- [ ] `tests/radtran/test_diagnostics.py::test_dofs_monotone_in_snr`
- [ ] `tests/radtran/test_quality.py::test_aod_ceiling_flag`

## Relationships
- Parent (theme epic): #73
- Blocked by: optimal-estimation iterative retrieval loop
- Blocks: Tier IV quality-flag aggregator
